app/Umbraco/Umbraco.Archetype/Events/StartUp.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Web.Configuration;
+using Umbraco.Core;
+
+namespace Archetype.Events
+{
+    public class StartUp : ApplicationEventHandler
+    {
+        protected override void ApplicationStarting(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
+        {
+            base.ApplicationStarting(umbracoApplication, applicationContext);
+
+            var config = WebConfigurationManager.OpenWebConfiguration("~");
```

Comment:
Using web config and updating it runtime is a bad idea. Both because it'll trigger a restart of the site (which is already in a stat-up context when this code is run), and because web configs are overwritten when deploying (which might cause the ID to be lost).

Can't we use a dedicated Archetype config in /config?

Reply:
@kgiszewski Does the GUID need to be stored? If you're after consistency, could do an MD5 of the hostname and GUID that?

Reply:
@kjac Great feedback. I'll definitely go the config file route to avoid the messy bits I have around the web.config.

@leekelleher Yes, consistency is what I'm after. I'm sort of stuck either way I go. I.e. a dev url and a prod url will generate different hashes. A config GUID will remain the same. But I see pros\cons to both.